xarray/tests/test_plot.py

```diff
@@ -1162,9 +1161,13 @@ def test_3d_raises_valueerror(self):
         with pytest.raises(ValueError, match=r"DataArray must be 2d"):
             self.plotfunc(a)
 
-    def test_nonnumeric_index_raises_typeerror(self):
+    def test_nonnumeric_index(self):
         a = DataArray(easy_array((3, 2)), coords=[["a", "b", "c"], ["d", "e"]])
-        with pytest.raises(TypeError, match=r"[Pp]lot"):
+        if self.plotfunc.__name__ == "surface":
+            # ax.plot_surface errors with nonnumerics:
+            with pytest.raises(np.core._exceptions._UFuncNoLoopError):
```

Comment:
```suggestion
            with pytest.raises(Exception):
```

so that we are not depending on an extremely private Error?